src/core/format-utils.js

```diff
@@ -0,0 +1,43 @@
+import BigNumber from 'bignumber.js';
+
+export const DOCK_TOKEN_UNIT = 1000000;
+
+export function formatCurrency(value, currency = 'USD') {
+  var formatter = new Intl.NumberFormat('en-US', {
+    style: 'currency',
+    currency: currency,
+
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 4,
+  });
+
+  return formatter.format(value);
+}
+
+export function formatAddress(value, size = 19) {
+  if (!value || size > value.length) {
+    return value;
+  }
+
+  const offset = size / 2;
+  return `${value.substring(0, offset)}...${value.substring(
+    value.length - offset,
+  )}`;
+}
+
+export function formatDockAmount(value) {
+  return BigNumber(value).dividedBy(DOCK_TOKEN_UNIT).toNumber();
```

Comment:
any danger of an undefined or invalid value throwing exceptions in here? And/or is that okay?

Reply:
No it will not throwing anything, BugNumber will handle undefined as NaN.

Reply:
what about invalid values? How would it handle `BigNumber("this is not a number")`?

Reply:
yeah, make sense. I've added a validation for it